dali/base/dasds.cpp

```diff
@@ -1429,24 +1429,26 @@ class CDeltaWriter : implements IThreaded
     }
     void flush()
     {
-        bool waitOnEmptySem = false;
+        // ensure all pending transactions are written out
+        // if necessary wait until CDeltaWriter thread has finished
+        bool allWritten = false;
         {
             CriticalBlock b(pendingCrit);
-            if (waiting)
+            if (writerIdle)
             {
-                if (pendingSz)
+                if (pendingSz) // the writer thread is idle, but there are some transactions that the writer hasn't seen yet
                     signalWaiting();
             }
-            if (!waiting)
+            if (!writerIdle) // i.e. if not idle, it is busy writing
```

Comment:
Worth having a comment that this includes if signalWaiting() was called above.  Otherwise it is very tempting to turn this if into an else, because it strange that writerIdle would be set by the requester.
writerRequested would be clearer than writerIdle, but would invert the condition, so safer not to make that change now.

signalWaiting() would similarly be better renamed as request[Async]Write()

Reply:
have made these changes, but kept writerIdle -> writeRequested change in it's own commit to allow easier review